distro/src/avail/Avail.avail/Foundation.avail/Lexers.avail

```diff
@@ -338,6 +341,110 @@ is
 	]
 ];
 
+/**
+ * Upon encountering an {@code "Arabic numeral"} digit, attempt to scan
+ * forward past all digit groups (optionally separated with single {@code "_"}
+ * digit group separator) until a single {@code "."} is encountered. Then
+ * require one or more digit groups. Allow an exponent signifier {@code "e"}
+ * before the final digit group, which may be prefixed with a sign. Allow an
+ * {@code "f"} in the final position to indicate that the value should be stored
+ * as a double.
+ *
+ * Note that a floating point number literal MUST have a digit in the integer
+ * part; 0 is acceptable.
+ *
+ * Regex: /[0-9]+(_[0-9]+)*\.([0-9]+_)*[0-9]+(e[+-]?([0-9]+_)*[0-9]+)?f?/
+ *
+ * @lexer "floating point number lexer"
+ * @param "source" "string"
+ *        The complete source text.
+ * @param "position" "real source position"
+ *        The position of the leading {@type "character"}.
+ * @param "line" "real source line number"
+ *        The line number of the leading {@type "character"}.
+ * @returns "{token+|}"
+ *          The possible lexes discovered at {@code "position"}, potentially
+ *          conforming to a single {@type "integer"} {@type "literal token"}.
+ */
```

Comment:
Fantastic documentation; it captures everything important.

Reply:
Agreed.  @leslieATavail and I tried out a few factorizations before settling on this one.

Reply:
Also tweaked line 351, since f indicates float, not double.